src/modules/Dashboard/features/Respondents/Popups/DataExportPopup/DataExportPopup.hooks.tsx

```diff
@@ -0,0 +1,146 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+import { exportDecryptedDataSucceed } from 'shared/utils/exportData';
+import { Mixpanel } from 'shared/utils/mixpanel/mixpanel';
+import { getParsedEncryptionFromServer } from 'shared/utils/encryption';
+import { ExportDataResult } from 'shared/types/answer';
+import { useEncryptionStorage } from 'shared/hooks/useEncryptionStorage';
+import { isProduction } from 'shared/utils/env';
+import { SessionStorageKeys } from 'shared/utils/storage';
+import { ItemResponseType } from 'shared/consts';
+
+import { IdleWorker, MultipleDecryptWorkersProps } from './DataExportPopup.types';
+import { WorkerOnMessageEvent } from './DataExportWorker/DataExportWorker.types';
+import DecryptionWorker from './DataExportWorker/DataExportWorker.worker';
+import { getExportDataSuffix } from './DataExportPopup.utils';
+import { NUM_WORKERS } from './DataExportWorker/DataExportWorker.const';
+
+export const useMultipleDecryptWorkers = ({
+  handleExportPopupClose,
+  appletId,
+  encryption,
+  filters,
+}: MultipleDecryptWorkersProps) => {
+  const [dataIsExporting, setDataIsExporting] = useState(false);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [exportDataQueue, setExportDataQueue] = useState<ExportDataResult[] | null>(null);
```

Comment:
Is there a reason to use `useState` for `exportDataQueue` and `currentPage`? None of the gets exported and used in the render

Reply:
`exportDataQueue` is used for storing the queue of received responses from the database. When it changes, the corresponding `useEffect` is triggered to process the next part of the queue. The part of `exportDataQueue` that needs processing is determined by `currentPage`. They are not meant to be used in the render process, but rather to manage side effects in the component.